src/popover/popover.jsx

```diff
@@ -0,0 +1,337 @@
+    className: React.PropTypes.string,
+    isOpen: React.PropTypes.bool,
+    onRequestClose: React.PropTypes.func,
+    style: React.PropTypes.object,
+    targetOrigin: PropTypes.origin,
+    zDepth: PropTypes.zDepth,
+  },
+
+  getDefaultProps() {
+    return {
+      anchorOrigin: {
+        vertical:'bottom',
+        horizontal:'left',
+      },
+      animated:true,
+      autoCloseWhenOffScreen:true,
+      canAutoPosition:true,
+      onRequestClose: () => {},
+      isOpen:false,
+      style: {},
+      targetOrigin: {
+        vertical:'top',
+        horizontal:'left',
+      },
+      zDepth: 1,
+    };
+  },
+
+  getInitialState() {
+    this.setPlacementThrottled = throttle(this.setPlacement, 100);
+    return {
+      open: false,
+    };
+  },
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  windowListeners: {
+    resize: 'setPlacementThrottled',
+    scroll: 'setPlacementThrottled',
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isOpen !== this.state.isOpen) {
+      if (nextProps.isOpen)
+        this._showInternal(nextProps.anchorEl);
+      else
+        this._hideInternal();
+    }
+  },
+
+  componentDidUpdate() {
+    this.setPlacement();
+  },
+
+  render() {
+    return <RenderToLayer
+      ref="layer"
+      {...this.props}
+      componentClickAway={this.componentClickAway}
+      render={this.renderLayer} />;
+  },
+
+  renderLayer() {
+    let {
+      animated,
+      targetOrigin,
+      className,
+      zDepth,
+    } = this.props;
+
+    const anchorEl = this.props.anchorEl || this.anchorEl;
+    let anchor = this.getAnchorPosition(anchorEl);
+    let horizontal = targetOrigin.horizontal.replace("middle", "vertical");
+
+    let wrapperStyle = {
+      position: 'fixed',
+      top: anchor.top, 
+      left: anchor.left,
+      zIndex: 20,
+      opacity:1,
+      overflow:'auto',
+      maxHeight:'100%',
+      transform:'scale(0,0)',
+      transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+      transition: animated ? Transitions.easeOut('500ms', ['transform', 'opacity']) : null,
+    };
+    wrapperStyle = this.mergeAndPrefix(wrapperStyle, this.props.style);
+
+    let horizontalAnimation = {
+      maxHeight:'100%',
+      overflowY:'auto',
+      transform:'scaleX(0)',
+      opacity:1,
+      transition: animated ? Transitions.easeOut('250ms', ['transform', 'opacity']): null,
+      transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+    };
+
+    let verticalAnimation = {
+      opacity:1,
+      transform:'scaleY(0)',
+      transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+      transition: animated ? Transitions.easeOut('500ms', ['transform', 'opacity']) : null,
+    }
+
+    return (
+      <Paper style={wrapperStyle} zDepth={zDepth} className={className} >
+        <div>
+          <div style={horizontalAnimation}>
+            <div style={verticalAnimation}>
+                {this.props.children}
```

Comment:
This is a side not. The children will be rendered with an `opacity: 0` event if the `isOpen: false` is passed to the Popover component. This is not good for perf. We would have to fix this later.